### ISSUE TITLE
Can we have persistent history?

### ISSUE THREAD

**Issue by mrbig033** [ranger/ranger#1741]
Sometimes I must close Ranger, and when I get back it would be useful for `history_go -1` go to the last `N` directories before I closed it. Is it currently possible?

#### Runtime Environment

- Operating system and version: MX Linux
- Terminal emulator and version: xfce4-terminal 0.8.3
- Python version: Python 3.7.4
- Ranger version/commit: 1.9.2
- Locale: Locale: en_US.UTF-8

#### Current Behavior

After reopening Ranger, `history_go -1` doesn't anywhere.

#### Expected Behavior

After reopening Ranger, `history_go -1` should go to the last location before closing Ranger.

#### Context

When I open Ranger, I don't always remember where I was or exactly what I was doing before closing. I know `ranger-cd` is useful in this situation, but it only stores a single location and then returns me to that location on the shell. It does not store history.

#### Possible Solutions

Make history persistent, with a configurable number of stored locations (similar do Vim's undo system). Or at least make `history_go -1` go to the last location before closing `Ranger` for the last time (if using `ranger-cd`, the only difference would be that Ranger would go to the last history from the shells current directory). It goes without saying that, in this context, `history_go 1` should also work accordingly.

#### Steps to reproduce
1. `$ cd` - go to the home directory
1. `$ ranger` -  open ranger (*not* ranger-cd)
3. Navigate to any `~/.config/ranger`
4. Reopen ranger
5. Hit whatever key you have bound to `history_go -1`.

**Comment by toonn** [comment]
Command history already persists. It seems reasonable to do the same with location history.

**Comment by mrbig033** [comment]
@toonn so you're agreeing with me, right? I'm dense/not my first language.

**Comment by toonn** [comment]
Yes, I'm saying that the command history (up arrow in console) already persists and I don't see why location history shouldn't. As long as its configurable to be ephemeral, for privacy reasons.

**Comment by mrbig033** [comment]
> Yes, I'm saying that the command history (up arrow in console) already persists and I don't see why location history shouldn't. As long as its configurable to be ephemeral, for privacy reasons.

Oh cool, that would be awesome.

**Comment by tapyu** [comment]
This functionality would be awesome, indeed.

**Comment by patrickaldis** [comment]
would love this!

**Comment by raphaelfournier** [comment]
Is there any progress on this? It was added to a milestone in 2021, but does not seem to have been implemented since. 
I too would love that! Especially, I'm using ranger with mutt to select attachments for emails, and I need to close and reopen ranger, so a shortcut to go to the previous folder visited *even before closing* would be awesome!

**Comment by toonn** [comment]
@raphaelfournier, not yet, take a stab at it!
